recaf-plugin/src/main/java/dev/xdark/recaf/plugin/RecafPlugin.java

```diff
@@ -0,0 +1,139 @@
+package dev.xdark.recaf.plugin;
+
+import dev.xdark.recaf.plugin.java.ZipPluginLoader;
+import me.coley.recaf.io.ByteSources;
+import me.coley.recaf.util.Directories;
+import me.coley.recaf.util.logging.Logging;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+
+/**
+ * @author xtherk
+ */
+public class RecafPlugin {
+    private static final Logger logger = Logging.get(RecafPlugin.class);
+    private final PluginManager pluginManager = new SimplePluginManager();
+    private final List<Path> pluginJarUrls = new ArrayList<>();
+    private final Map<PluginContainer<? extends Plugin>, Path> pluginContainerPathMap = new HashMap<>();
+    private static RecafPlugin INSTANCE;
+    private static boolean initialized = false;
+    private static final String[] SUFFIX = {".jar"};
+
+    {
+        pluginManager.registerLoader(new ZipPluginLoader(RecafPlugin.class.getClassLoader()));
```

Comment:
Any reason to not put this in the `private` constructor below?

Reply:
no problem